### PR TITLE
BUG: series with complex nan

### LINKED ISSUE #53627: BUG: pd.Series([complex('nan')]) raises
```python
>>> pd.Series([complex('nan')])
[...]
  File "lib.pyx", line 2488, in pandas._libs.lib.maybe_convert_objects
TypeError: must be real number, not complex
```

### PR DESCRIPTION
**Pull request by Charlie-XIAO**
- [x] Closes #53627
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests)
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit)
- [x] Added an entry in the latest `doc/source/whatsnew/v2.1.0.rst` file

### CODE REVIEW

File: pandas/tests/generic/test_series.py [hunk 1]
```diff
@@ -157,3 +157,8 @@ def finalize(self, other, method=None, **kwargs):
             result = pd.concat([ser, ser2])
             assert result.filename == "foo+bar"
             assert result.name is None
+
+    def test_series_with_complex_nan(self):
+        # GH#53627
+        ser = Series([complex("nan")])
```
Comment (jbrockmendel) [inline pandas/tests/generic/test_series.py:163]: can you also check `ser = Series([complex('1+j'), complex('nan'), complex('2+2j')])`?  thats the actual case that i first stumbled on

File: pandas/tests/generic/test_series.py [hunk 2]
```diff
@@ -157,3 +157,8 @@ def finalize(self, other, method=None, **kwargs):
             result = pd.concat([ser, ser2])
             assert result.filename == "foo+bar"
             assert result.name is None
+
+    def test_series_with_complex_nan(self):
```
Comment (jbrockmendel) [inline pandas/tests/generic/test_series.py:161]: this should go in tests.series.test_construction

**Comment by Charlie-XIAO** [comment]
@jbrockmendel suggested changes made, thanks for your review!

One thing to notice is the repr:
```
>>> import pandas as pd
>>> pd.Series([complex("nan")])
0   NaN
dtype: complex128
>>> pd.Series([1, complex("nan"), 2])
0    1.0+0.0j
1   N000a000N
2    2.0+0.0j
dtype: complex128
```
I don't think this is as expected, but I suppose this should be a separate issue?

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @Charlie-XIAO 

> I don't think this is as expected, but I suppose this should be a separate issue?

Yeah that appears to be a bug. Could you open a separate issue about that?

**Comment by Charlie-XIAO** [comment]
Sure, will do soon.

**Pull request merged**